### PR TITLE
Import package version from package.json

### PR DESCRIPTION
**Pull request by mmermerkaya**
Imports version strings directly from `package.json`, which allows TS to do type checking and warn if there's a typo or a type mismatch. Removes the need for `rollup-plugin-replace`.

### CODE REVIEW

File: packages/app/test/firebaseApp.test.ts [hunk 1]
```diff
@@ -15,6 +15,8 @@
  * limitations under the License.
  */
 
+import './setup';
```
Comment (dwoffinden) [inline packages/app/test/firebaseApp.test.ts]: :/

File: packages/util/src/global.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/** The global object in both Node and browser environments. */
+export const root: any =
+  (typeof self === 'object' && self.self === self && self) ||
```
Comment (dwoffinden) [inline packages/util/src/global.ts]: bonkers

File: packages/app/rollup.config.js [hunk 3]
```diff
@@ -34,9 +34,9 @@ const es5BuildPlugins = [
     typescript
   }),
   replace({
-    delimiters: ['${', '}'],
+    delimiters: ['__', '__'],
     values: {
-      JSCORE_VERSION: firebasePkg.version
+      JSCORE_VERSION: JSON.stringify(firebasePkg.version)
```
[the remark this answers is not in the document]
Comment (mmermerkaya) [reply packages/app/rollup.config.js]: Without stringify, the replacement would be the value of the string, as `x.y.z`. So the code would look like this: `const version = 1.2.3;`

With stringify, it's instead replaced with `"x.y.z"`, so the example would be `const version = "1.2.3";`, which is what we want.

Rollup plugin example here: https://github.com/rollup/rollup-plugin-replace#usage

File: packages/app/src/lite/firebaseNamespaceLite.ts [hunk 4]
```diff
@@ -28,7 +28,7 @@ import { createFirebaseNamespaceCore } from '../firebaseNamespaceCore';
 export function createFirebaseNamespaceLite(): FirebaseNamespace {
   const namespace = createFirebaseNamespaceCore(FirebaseAppLiteImpl);
 
-  namespace.SDK_VERSION = '${JSCORE_VERSION}_LITE';
+  namespace.SDK_VERSION = `${__JSCORE_VERSION__}_LITE`;
```
[the remark this answers is not in the document]
Comment (mmermerkaya) [reply packages/app/src/lite/firebaseNamespaceLite.ts]: I didn't want to change the original.

**Comment by mmermerkaya** [comment]
Yeah I agree about the testing stuff. I didn't realize that it would take so much effort to make it work until it was too late :sweat_smile:

I like your version, thanks for that! I tried to keep the replace plugin in case we use it for something else, but all three packages that I touched only use it for the version string. YAGNI, I guess :smile:

I'll merge your PR to my branch, and replace the others as well.

**Comment by mmermerkaya** [comment]
Done. Edited OP as well.

**Comment by mmermerkaya** [comment]
Looks like this changed the TypeScript output directory structure in App and Installations, because we added an import from a parent directory of the current top level directory in both of these packages. Should be fine with the `.d.ts` file location fixes in `package.json`s, but it would be great if we could use something like [dts-bundle-generator](https://github.com/timocov/dts-bundle-generator) or [rollup-plugin-dts](https://github.com/Swatinem/rollup-plugin-dts) to generate a `d.ts` bundle instead of relying on declarations generated by the TS compiler, just like we do with the actual code.

**Pull request merged**